src/logger.ts

```diff
@@ -156,25 +171,28 @@ export function isDebugEnabled(): boolean {
   return getLogLevel() === 'debug';
 }
 
+/**
+ * Creates a logger method for the specified log level
+ */
+function createLogMethod(level: keyof typeof LOG_LEVELS): StrictLogMethod {
```

Comment:
IMO the previous format is preferable OR don't use this function for `debug`.

Reply:
I do see your point. I was trying to DRY off. This is handling a complex requirement - debug level should always include location, and other levels should include location only when debug is enabled.